--- origsrc/nickle-2.68/Makefile.am	2008-06-21 04:22:32.000000000 -0500
+++ src/nickle-2.68/Makefile.am	2009-02-26 00:09:31.513625000 -0600
@@ -18,8 +18,9 @@ DEBIAN = debian/nickle.install debian/ch
 	 debian/control debian/copyright debian/rules debian/lintian.override
 	 
 EXTRA_DIST = README.name README.release autogen.sh ChangeLog \
-	$(NICKLEFILES) $(DEBIAN) nickle.1.in nickle.spec.in
+	$(NICKLEFILES) $(DEBIAN) nickle.1.in nickle.pc.in nickle.spec.in
 
+DISTCLEANFILES = nickle.pc
 MAINTAINERCLEANFILES=ChangeLog
 
 .PHONY: ChangeLog
@@ -30,6 +31,9 @@ ChangeLog:
 
 dist-hook: ChangeLog
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = nickle.pc
+
 nickleincludedir=$(includedir)/nickle
 
 nickleinclude_HEADERS = \
@@ -53,7 +57,7 @@ nickle_SOURCES = \
 	builtin.c builtin.h \
 	builtin-foreign.c gram.y lex.l
 
-nickle_LDFLAGS=$(NICKLE_LDFLAGS)
+nickle_LDFLAGS=$(NICKLE_LDFLAGS) -Wl,--export-all,--out-implib,libnickle.a
 
 pkgdata_DATA = $(NICKLEFILES) COPYING
 
@@ -126,6 +130,10 @@ nickle.spec: nickle.spec.in config.h
 clean-local:
 	rm -f nickle.1 nickle.spec
 
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libnickle.a $(DESTDIR)$(libdir)/libnickle.a
+
 #
 # This assumes you've got Mike Harris's rpmbuild-nonroot stuff installed
 # using the defaults
--- origsrc/nickle-2.68/configure.in	2008-06-21 04:14:01.000000000 -0500
+++ src/nickle-2.68/configure.in	2009-02-26 00:09:31.529250000 -0600
@@ -131,6 +131,7 @@ AC_CONFIG_FILES(
  examples/Makefile
  examples/smlng/Makefile
  examples/turtle/Makefile
+ nickle.pc
  date-sh)
 
 AC_OUTPUT
--- origsrc/nickle-2.68/file.c	2008-03-27 01:16:02.000000000 -0500
+++ src/nickle-2.68/file.c	2009-02-27 07:52:31.592875000 -0600
@@ -20,9 +20,9 @@
 #ifdef O_ASYNC
 #define ASYNC O_ASYNC
 #else
+#define ASYNC 0
 #ifdef HAVE_STROPTS_H
 #define USE_STREAMS_ASYNC
-#define ASYNC 0
 #include <stropts.h>
 #endif
 #endif
--- origsrc/nickle-2.68/nickle.pc.in	1969-12-31 18:00:00.000000000 -0600
+++ src/nickle-2.68/nickle.pc.in	2009-02-26 00:09:31.529250000 -0600
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+includedir=@includedir@
+libdir=@libdir@
+nicklelibdir=@nicklelibdir@
+
+Name: Nickle
+Description: Nickle programming language
+Version: @VERSION@
+Libs: -L${libdir} -lnickle
+Cflags: -I${includedir}
